arch/arm/core/cortex_m/Kconfig

```diff
@@ -107,6 +107,16 @@ config ARMV7_M_ARMV8_M_MAINLINE
 	  The Main Extension provides backwards compatibility
 	  with ARMv7-M.
 
+config ARMV7_M_ARMV8_M_FP
```

Comment:
Do we really need this, isn't CONFIG_FLOAT sufficient?

Reply:
@galak Thanks for your comment. I think it is a nice to have feature because:

CONFIG_FLOAT is a _user_ option, and as such, it may be used or not used by the application (of course it depends on CPU_HAS_FPU).
However, even if it is _not_ enabled by the application, if FPU is present, it is always nice to be able to track down such errors. (For example, in case some nasty code has enabled FPU outside CONFIG_FLOAT, and these lazy state preservation errors pop-up somehow...)

Reply:
Than I think we should replace these with CPU_HAS_FPU